### DISCUSSION TITLE
ray tracing compilation error (constexpr mixedPackSize())

### DISCUSSION THREAD

**Discussion by WilkAndy** [idaholab/moose#17576] [Q&A General] [answered]
Hey everyone,

Any hints on how to fix this:
```
ray_tracing/build/header_symlinks/RayTracingPackingUtils.h:211:1: note: ‘constexpr std::size_t RayTracingPackingUtils::mixedPackSize() [with BufferType = double; InputTypes = {short unsigned int, bool, unsigned int, unsigned int, unsigned int}; std::size_t = long unsigned int]’ is not usable as a constexpr function because:
 mixedPackSize()
 ^
/datasets/work/en-cmrp/work/apps/projects/projects_wil04q/moose/framework/build.mk:122: recipe for target '/datasets/work/en-cmrp/work/apps/projects/projects_wil04q/moose/modules/ray_tracing/build/unity_src/raytracing_Unity.x86_64-pc-linux-gnu.dbg.lo' failed
make: *** [/datasets/work/en-cmrp/work/apps/projects/projects_wil04q/moose/modules/ray_tracing/build/unity_src/raytracing_Unity.x86_64-pc-linux-gnu.dbg.lo] Error 1
```

a

**Comment by loganharbour** [comment, marked as answer]
This is a duplicate of #17568. Continuing conversation over there.

Resolution:
https://github.com/idaholab/moose/discussions/17568#discussioncomment-601085

**Reply by loganharbour** [reply]
@WilkAndy I'm glad to help with getting your compile going:

- If you're using conda libMesh, you'll need to temporarily cherry-pick the first commit in https://github.com/idaholab/moose/pull/17582, which is https://github.com/idaholab/moose/pull/17582/commits/6c5399414eee32308487e0da3d46612344a4f794
- If you're not using conda libMesh, you'll need to recompile libMesh by introducing the change to `scripts/update_and_rebuild_libmesh.sh` in https://github.com/idaholab/moose/pull/17581/commits/40bdca04ac57d9a1206675d0646f6edd6ac5addc 

We're in the middle of system maintenance so either one of those PRs won't be merged in the near future.

**Reply by WilkAndy** [reply]
Thanks @loganharbour .   Probably this works.  Unfortunately my HPC's compiler is not C++14 .    Sigh.... i bet this is going to be a saga.
